diff --git a/node_modules/react-native-tab-view/lib/commonjs/TabBar.js b/node_modules/react-native-tab-view/lib/commonjs/TabBar.js
index 9479ff6..f46384a 100644
--- a/node_modules/react-native-tab-view/lib/commonjs/TabBar.js
+++ b/node_modules/react-native-tab-view/lib/commonjs/TabBar.js
@@ -303,9 +303,10 @@ function TabBar(_ref9) {
       defaultTabWidth: !isWidthDynamic ? getComputedTabWidth(index, layout, routes, scrollEnabled, tabWidths, getFlattenedTabWidth(tabStyle)) : undefined,
       android_ripple
     };
-    return /*#__PURE__*/React.createElement(React.Fragment, null, gap > 0 && index > 0 ? /*#__PURE__*/React.createElement(Separator, {
+    const { key: itemKey, ...itemProps } = props;
+    return /*#__PURE__*/React.createElement(React.Fragment, { key: itemKey }, gap > 0 && index > 0 ? /*#__PURE__*/React.createElement(Separator, {
       width: gap
-    }) : null, renderTabBarItem ? renderTabBarItem(props) : /*#__PURE__*/React.createElement(_TabBarItem.TabBarItem, props));
+    }) : null, renderTabBarItem ? renderTabBarItem(props) : /*#__PURE__*/React.createElement(_TabBarItem.TabBarItem, { key: itemKey, ...itemProps }));
   }, [activeColor, android_ripple, gap, getAccessibilityLabel, getAccessible, getLabelText, getTestID, inactiveColor, isWidthDynamic, jumpTo, labelStyle, layout, navigationState, onTabLongPress, onTabPress, position, pressColor, pressOpacity, renderBadge, renderIcon, renderLabel, renderTabBarItem, routes, scrollEnabled, tabStyle, tabWidths]);
   const keyExtractor = React.useCallback(item => item.key, []);
   const contentContainerStyleMemoized = React.useMemo(() => [styles.tabContent, scrollEnabled ? {
diff --git a/node_modules/react-native-tab-view/lib/module/TabBar.js b/node_modules/react-native-tab-view/lib/module/TabBar.js
index 2714896..4658433 100644
--- a/node_modules/react-native-tab-view/lib/module/TabBar.js
+++ b/node_modules/react-native-tab-view/lib/module/TabBar.js
@@ -294,9 +294,10 @@ export function TabBar(_ref9) {
       defaultTabWidth: !isWidthDynamic ? getComputedTabWidth(index, layout, routes, scrollEnabled, tabWidths, getFlattenedTabWidth(tabStyle)) : undefined,
       android_ripple
     };
-    return /*#__PURE__*/React.createElement(React.Fragment, null, gap > 0 && index > 0 ? /*#__PURE__*/React.createElement(Separator, {
+    const { key: itemKey, ...itemProps } = props;
+    return /*#__PURE__*/React.createElement(React.Fragment, { key: itemKey }, gap > 0 && index > 0 ? /*#__PURE__*/React.createElement(Separator, {
       width: gap
-    }) : null, renderTabBarItem ? renderTabBarItem(props) : /*#__PURE__*/React.createElement(TabBarItem, props));
+    }) : null, renderTabBarItem ? renderTabBarItem(props) : /*#__PURE__*/React.createElement(TabBarItem, { key: itemKey, ...itemProps }));
   }, [activeColor, android_ripple, gap, getAccessibilityLabel, getAccessible, getLabelText, getTestID, inactiveColor, isWidthDynamic, jumpTo, labelStyle, layout, navigationState, onTabLongPress, onTabPress, position, pressColor, pressOpacity, renderBadge, renderIcon, renderLabel, renderTabBarItem, routes, scrollEnabled, tabStyle, tabWidths]);
   const keyExtractor = React.useCallback(item => item.key, []);
   const contentContainerStyleMemoized = React.useMemo(() => [styles.tabContent, scrollEnabled ? {
